### ISSUE TITLE
Break down the API Documentation page

### ISSUE THREAD

**Issue by cool-RR** [spatialaudio/python-sounddevice#205]
I'm using the API Documentation page and its length is make it difficult to use. Streams are a complicated subject, and it doesn't help that their documentation is on the same page with the other features.

In the beginning of the page you have an "API overview" section with 7 bullets, I suggest that the division to pages be roughly as that of the bullets.

**Comment by mgeier** [comment]
That's a great idea, would you like to make a PR for that?

The main API page could have this (without `:members:`!):

    .. automodule:: sounddevice

And then a `toctree` directive with the sub-pages.

Most of [the current API page](../blob/afe8fcc4134fe2b9a5f89c15c76fc2b67c15a3d4/doc/api.rst) is already split into separate pieces, so it should be easy to create separate pages.

**Comment by cool-RR** [comment]
Can you please give me the code needed to build the documentation locally?

I tried this snippet that I found in `.travis.yml`:

```
python setup.py build_sphinx -b linkcheck -W
```

But I got this error:

```
C:\Users\Administrator\Documents\Python Projects\Third-party\sounddevice\doc\index.rst:5:toctree conta
ins reference to document 'CONTRIBUTING' that doesn't have a title: no link will be generated         
```

**Comment by mgeier** [comment]
The command you used doesn't actually build the HTML files, it only checks if all the links are still valid.

Here are the instructions for building the docs: https://github.com/spatialaudio/python-sounddevice/blob/master/CONTRIBUTING.rst

**Comment by mgeier** [comment]
This was fixed in #207.
